client/components/tinymce/plugins/simple-payments/dialog/index.jsx

```diff
@@ -161,16 +161,23 @@ class SimplePaymentsDialog extends Component {
 	// or the default values for a new one.
 	getInitialFormFields( paymentId ) {
 		const { initialFields } = this.constructor;
+		const { paymentButtons, currentUserEmail } = this.props;
 
 		if ( isNumber( paymentId ) ) {
-			const editedPayment = find( this.props.paymentButtons, p => p.ID === paymentId );
+			const editedPayment = find( paymentButtons, p => p.ID === paymentId );
 			if ( editedPayment ) {
 				// Pick only the fields supported by the form -- drop the rest
 				return pick( editedPayment, Object.keys( initialFields ) );
 			}
 		}
 
-		return initialFields;
+		const lastPaymentButtonEmail = get( paymentButtons, '0.email', null );
+
+		if ( ! paymentButtons || ! paymentButtons.length ) {
+			return { ...initialFields, email: currentUserEmail || lastPaymentButtonEmail };
+		}
+
+		return { ...initialFields, email: lastPaymentButtonEmail };
```

Comment:
Could this be just:
```js
const initialEmail = get( paymentButtons, '0.email', currentUserEmail );
return { ...initialFields, email: initialEmail };
```
?
`_.get` should take care of `null` or empty `paymentButtons` array and return the default value where needed.